drivers/regulator/regulator_rpi_pico.c

```diff
@@ -0,0 +1,185 @@
+		 (idx << VREG_AND_CHIP_RESET_VREG_VSEL_LSB));
+
+	return 0;
+}
+
+static int regulator_rpi_pico_get_voltage(const struct device *dev, int32_t *volt_uv)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *vreg = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->vreg);
+
+	return linear_range_group_get_value(
+		config->ranges, config->num_ranges,
+		((*vreg & VREG_AND_CHIP_RESET_VREG_VSEL_BITS) >> VREG_AND_CHIP_RESET_VREG_VSEL_LSB),
+		volt_uv);
+}
+
+static int regulator_rpi_pico_enable(const struct device *dev)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *vreg = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->vreg);
+
+	*vreg |= BIT(VREG_AND_CHIP_RESET_VREG_EN_LSB);
+
+	return 0;
+}
+
+static int regulator_rpi_pico_disable(const struct device *dev)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *vreg = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->vreg);
+
+	*vreg &= ~BIT(VREG_AND_CHIP_RESET_VREG_EN_LSB);
+
+	return 0;
+}
+
+static int regulator_rpi_pico_set_mode(const struct device *dev, regulator_mode_t mode)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *vreg = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->vreg);
+
+	if (mode & REGULATOR_RPI_PICO_MODE_HI_Z) {
+		*vreg |= REGULATOR_RPI_PICO_MODE_HI_Z;
+	} else {
+		*vreg &= (~REGULATOR_RPI_PICO_MODE_HI_Z);
+	}
+
+	return 0;
+}
+
+static int regulator_rpi_pico_get_mode(const struct device *dev, regulator_mode_t *mode)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *vreg = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->vreg);
+
+	*mode = (*vreg & REGULATOR_RPI_PICO_MODE_HI_Z);
+
+	return 0;
+}
+
+static int regulator_rpi_pico_init(const struct device *dev)
+{
+	const struct regulator_rpi_pico_config *config = dev->config;
+	volatile uint32_t *bod = &(((volatile vreg_and_chip_reset_hw_t *)config->reg)->bod);
+
+	if (config->brown_out_detection) {
+		*bod = (BIT(VREG_AND_CHIP_RESET_BOD_EN_LSB) |
+			(config->brown_out_threshold << VREG_AND_CHIP_RESET_BOD_VSEL_LSB));
+	} else {
+		*bod &= ~BIT(VREG_AND_CHIP_RESET_BOD_EN_LSB);
+	}
+
+	regulator_common_data_init(dev);
+
+	return regulator_common_init(dev, true);
+}
+
+static int regulator_rpi_pico_common_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+static const struct regulator_driver_api api = {
+	.enable = regulator_rpi_pico_enable,
+	.disable = regulator_rpi_pico_disable,
+	.count_voltages = regulator_rpi_pico_count_voltages,
+	.list_voltage = regulator_rpi_pico_list_voltage,
+	.set_voltage = regulator_rpi_pico_set_voltage,
+	.get_voltage = regulator_rpi_pico_get_voltage,
+	.set_mode = regulator_rpi_pico_set_mode,
+	.get_mode = regulator_rpi_pico_get_mode,
+};
+
+static const struct regulator_driver_api parent_api = {
+	/* place holder */
+};
```

Comment:
note: parent device is not required, just drop it if not needed.

Reply:
Delete the parent device and make a flattened structure.